gax-grpc/src/main/java/com/google/api/gax/grpc/DirectStreamingCallable.java

```diff
@@ -29,71 +29,85 @@
   }
 
-  void serverStreamingCall(
-      RequestT request, ApiStreamObserver<ResponseT> responseObserver, CallContext context) {
+  @Override
+  public void serverStreamingCall(
+      RequestT request, ApiStreamObserver<ResponseT> responseObserver, ApiCallContext context) {
     Preconditions.checkNotNull(request);
     Preconditions.checkNotNull(responseObserver);
-    ClientCall<RequestT, ResponseT> call =
-        factory.newCall(context.getChannel(), context.getCallOptions());
+    ClientCall<RequestT, ResponseT> call = newCall(context);
     ClientCalls.asyncServerStreamingCall(
-        call, request, new RpcStreamObserverDelegate<ResponseT>(responseObserver));
+        call, request, new ApiStreamObserverDelegate<ResponseT>(responseObserver));
   }
 
-  Iterator<ResponseT> blockingServerStreamingCall(RequestT request, CallContext context) {
+  @Override
+  public Iterator<ResponseT> blockingServerStreamingCall(RequestT request, ApiCallContext context) {
     Preconditions.checkNotNull(request);
-    ClientCall<RequestT, ResponseT> call =
-        factory.newCall(context.getChannel(), context.getCallOptions());
+    ClientCall<RequestT, ResponseT> call = newCall(context);
     return ClientCalls.blockingServerStreamingCall(call, request);
   }
 
-  ApiStreamObserver<RequestT> bidiStreamingCall(
-      ApiStreamObserver<ResponseT> responseObserver, CallContext context) {
+  @Override
+  public ApiStreamObserver<RequestT> bidiStreamingCall(
+      ApiStreamObserver<ResponseT> responseObserver, ApiCallContext context) {
     Preconditions.checkNotNull(responseObserver);
-    ClientCall<RequestT, ResponseT> call =
-        factory.newCall(context.getChannel(), context.getCallOptions());
+    ClientCall<RequestT, ResponseT> call = newCall(context);
     return new StreamObserverDelegate<RequestT>(
         ClientCalls.asyncBidiStreamingCall(
-            call, new RpcStreamObserverDelegate<ResponseT>(responseObserver)));
+            call, new ApiStreamObserverDelegate<ResponseT>(responseObserver)));
   }
 
-  ApiStreamObserver<RequestT> clientStreamingCall(
-      ApiStreamObserver<ResponseT> responseObserver, CallContext context) {
+  @Override
+  public ApiStreamObserver<RequestT> clientStreamingCall(
+      ApiStreamObserver<ResponseT> responseObserver, ApiCallContext context) {
     Preconditions.checkNotNull(responseObserver);
-    ClientCall<RequestT, ResponseT> call =
-        factory.newCall(context.getChannel(), context.getCallOptions());
+    ClientCall<RequestT, ResponseT> call = newCall(context);
     return new StreamObserverDelegate<RequestT>(
         ClientCalls.asyncClientStreamingCall(
-            call, new RpcStreamObserverDelegate<ResponseT>(responseObserver)));
+            call, new ApiStreamObserverDelegate<ResponseT>(responseObserver)));
+  }
+
+  public ClientCall<RequestT, ResponseT> newCall(ApiCallContext context) {
+    if (!(context instanceof GrpcCallContext)) {
```

Comment:
The types `UnaryCallable`, `OperationCallable`, and `StreamingCallable` will be present on the transport-agnostic stub, and thus can't be parameterized with the transport-specific context.

In order to have a stub that can be backed by any transport, there needs to be some entry points that are unsupported by some transports, so the `StreamingCallable` still needs to be present there, even if we only anticipate grpc supporting it.